tests/models/layoutlmv3/test_modeling_tf_layoutlmv3.py

```diff
@@ -0,0 +1,497 @@
+            _,
+            _,
+        ) = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_model(config, input_ids, bbox, pixel_values, token_type_ids, input_mask)
+
+    def test_model_various_embeddings(self):
+        (
+            config,
+            input_ids,
+            bbox,
+            pixel_values,
+            token_type_ids,
+            input_mask,
+            _,
+            _,
+        ) = self.model_tester.prepare_config_and_inputs()
+        for type in ["absolute", "relative_key", "relative_key_query"]:
+            config.position_embedding_type = type
+            self.model_tester.create_and_check_model(config, input_ids, bbox, pixel_values, token_type_ids, input_mask)
+
+    def test_for_sequence_classification(self):
+        (
+            config,
+            input_ids,
+            bbox,
+            pixel_values,
+            token_type_ids,
+            input_mask,
+            sequence_labels,
+            _,
+        ) = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_sequence_classification(
+            config, input_ids, bbox, pixel_values, token_type_ids, input_mask, sequence_labels
+        )
+
+    def test_for_token_classification(self):
+        (
+            config,
+            input_ids,
+            bbox,
+            pixel_values,
+            token_type_ids,
+            input_mask,
+            _,
+            token_labels,
+        ) = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_token_classification(
+            config, input_ids, bbox, pixel_values, token_type_ids, input_mask, token_labels
+        )
+
+    def test_for_question_answering(self):
+        (
+            config,
+            input_ids,
+            bbox,
+            pixel_values,
+            token_type_ids,
+            input_mask,
+            sequence_labels,
+            _,
+        ) = self.model_tester.prepare_config_and_inputs()
+        self.model_tester.create_and_check_for_question_answering(
+            config, input_ids, bbox, pixel_values, token_type_ids, input_mask, sequence_labels
+        )
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_name in TF_LAYOUTLMV3_PRETRAINED_MODEL_ARCHIVE_LIST[:1]:
+            model = TFLayoutLMv3Model.from_pretrained(model_name)
+            self.assertIsNotNone(model)
+
+
+# We will verify our results on an image of cute cats
+def prepare_img():
+    image = Image.open("./tests/fixtures/tests_samples/COCO/000000039769.png")
+    return image
+
+
+@require_tf
+class TFLayoutLMv3ModelIntegrationTest(unittest.TestCase):
+    @cached_property
+    def default_feature_extractor(self):
+        return LayoutLMv3FeatureExtractor(apply_ocr=False) if is_vision_available() else None
+
+    @slow
+    def test_inference_no_head(self):
+        model = TFLayoutLMv3Model.from_pretrained("microsoft/layoutlmv3-base")
```

Comment:
Does this work out of the box? Do we need to upload the TF checkpoits here?

Reply:
No it does not work out of the box. It will work when the TF checkpoints are uploaded there though. I have checked that the test passes when setting `from_pt=True`. The same goes for `test_model_from_pretrained`.

The TF checkpoints need to be uploaded to:
* microsoft/layoutlmv3-base
* microsoft/layoutlmv3-large

Possibly also to some of the fine-tuned models?